app/services/caution_decision_tree.rb

```diff
@@ -1,17 +1,33 @@
 class CautionDecisionTree < BaseDecisionTree
+  # rubocop:disable Metrics/CyclomaticComplexity
   def destination
     return next_step if next_step
 
     case step_name
+    when :caution_type
```

Comment:
Although in terms of code it makes no difference, it is easier to understand the decision flow if we maintain the steps ordered. So in this case, this should go at the end.